### DISCUSSION TITLE
Purpose of tmp directory in repository?

### DISCUSSION THREAD

**Discussion by SirWumpus** [ioccc-src/mkiocccentry#889] [Q&A] [answered]
What is the purpose of the `tmp` directory in the repository?  Shouldn't working files and generated files be excluded from the repository?  When someone see a `tmp` directory they might be inclined to delete its contents and/or the whole directory.  If its some sort of intermediate build directory, maybe renaming it to something better suited, eg. `build`, `stuff`, `wip` (work in progress), ...

- - -
If its source that is part of the `soup`, then should it not live in `soup`?  Or at least reflect its purpose, eg. `json_tools`, `json_utils` ?

**Comment by lcn2** [comment, marked as answer]
The contents of the tmp directory should not be part of the repo. 

It contains some fragments that might be part of a future release.

I suggest that we put the tmp directory in `.gitignore`.

**Comment by lcn2** [comment]
Removed the tmp directory with commit 5658711b7750e18b6bba79249fd9dbfcfa5e6838.

**Comment by xexyl** [comment]
As for the json tools the ones that matter are in `jparse`. But as discussed elsewhere three of them will be mostly rewritten. Those are the reason for the hacks in the other repo.

Of course this is late in the coming but perhaps you'd like to know where the jparse tools will be. Actually it'll be in another repo like the dbg library/repo and dyn_array library/repo. But that will come later and will take a bit of work (hence later). Those in `tmp/` were once thought necessary but that was changed. Perhaps it was thought it might be needed again but I don't really know.
